syft/pkg/cataloger/common/cpe/generate.go

```diff
@@ -129,6 +133,8 @@ func candidateVendors(p pkg.Package) []string {
 	// generate sub-selections of each candidate based on separators (e.g. jenkins-ci -> [jenkins, jenkins-ci])
 	addAllSubSelections(vendors)
 
+	vendors.removeByCondition(invalidCPEValue)
```

Comment:
That's interesting! I'm up for that approach. If I understand correctly: before returning all generated CPEs, we'd test each one by serializing it and then deserializing it. Is that right?

@wagoodman Curious for your thoughts here, too

Reply:
... I just remembered we have benchmark tests that could show us a substantial change in performance!

Reply:
Attempted in 1a51bd5 — let me know what you think

Reply:
I like this approach much better... instead of covering the single case of `:` we're getting to the root problem of "don't allow CPEs that will never work" 👍